sway/commands/layout.c

```diff
@@ -49,11 +49,86 @@ struct cmd_results *cmd_layout(int argc, char **argv) {
 		} else if (strcasecmp(argv[0], "splitv") == 0) {
 			swayc_change_layout(parent, L_VERT);
 		} else if (strcasecmp(argv[0], "toggle") == 0 && argc == 2 && strcasecmp(argv[1], "split") == 0) {
-			if (parent->layout == L_HORIZ && (parent->workspace_layout == L_NONE || parent->workspace_layout == L_HORIZ)) {
+			if (parent->layout == L_HORIZ && (parent->workspace_layout == L_NONE ||
+							  parent->workspace_layout == L_HORIZ)) {
 				swayc_change_layout(parent, L_VERT);
 			} else {
 				swayc_change_layout(parent, L_HORIZ);
 			}
+		} else if (strcasecmp(argv[0], "auto_left") == 0) {
+			swayc_change_layout(parent, L_AUTO_LEFT);
+		} else if (strcasecmp(argv[0], "auto_right") == 0) {
+			swayc_change_layout(parent, L_AUTO_RIGHT);
+		} else if (strcasecmp(argv[0], "auto_top") == 0) {
+			swayc_change_layout(parent, L_AUTO_TOP);
+		} else if (strcasecmp(argv[0], "auto_bot") == 0) {
+			swayc_change_layout(parent, L_AUTO_BOTTOM);
+		} else if (strcasecmp(argv[0], "incnmaster") == 0)  {
+			if ((error = checkarg(argc, "layout incnmaster",
+					      EXPECTED_EQUAL_TO, 2))) {
+				return error;
+			}
+			int inc = (int) strtol(argv[1], NULL, 10);
```

Comment:
Can you add error handling for invalid ints here?

```c
char *end;
int inc = (int)strtol(argv[1], &end, 10);
if (*end) {
    // error
}
```

Also note no space between (int) and strtol